### ISSUE TITLE
bazel: `TestLint/TestGCAssert` fails when run under Bazel

### ISSUE THREAD

**Issue by rickystewart** [cockroachdb/cockroach#65485]
We see a lot of error output like this:

```
...
        ../../sql/colexec/sort.eg.go:2731:	s.sortCol.Get(s.order[j]): call was not inlined
        ../../sql/colexec/sort.eg.go:2921:	s.sortCol.Get(s.order[i]): call was not inlined
        ../../sql/colexec/sort.eg.go:2922:	s.sortCol.Get(s.order[j]): call was not inlined
        ../../sql/colexec/sort.eg.go:2990:	s.sortCol.Get(s.order[i]): call was not inlined
        ../../sql/colexec/sort.eg.go:2991:	s.sortCol.Get(s.order[j]): call was not inlined
        ../../sql/colexec/sort.eg.go:3051:	s.sortCol.Get(s.order[i]): call was not inlined
...
```

The same errors don't appear when the lint test is run outside the Bazel context. We're not sure if these errors are spurious or false alarms.

Epic CRDB-8349
Jira issue: CRDB-7629

**Comment by rickystewart** [comment]
Not actually done, this job was flaky in TC and is skipped.

**Comment by srosenberg** [comment]
> Not actually done, this job was flaky in TC and is skipped.

TL;DR The reason it's failing under bazel is due to the missing `./dev gen cgo` step.

#### Reproducing

Let's use the PR [1] which captures "go build" output,

```
/home/srosenberg/go/src/github.com/gcassert/gcassert $(cat ./pkg/testutils/lint/gcassert_paths.txt | sed 's|^|./pkg/|')
See /tmp/gcassert-329905986.log for full output.
exit status 1
```

Upon closer examination of `/tmp/gcassert-329905986.log`,

```
pkg/server/status/runtime_jemalloc.go:21:11: fatal error: 'jemalloc/jemalloc.h' file not found
 #include <jemalloc/jemalloc.h>
          ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

Thus, during type-checking a native (transitive) dependency is unresolved, causing `go build` to bail out early.

#### Fixing

Generate zcgo files,

```
./dev gen go cgo
```

Run,

```
/home/srosenberg/go/src/github.com/gcassert/gcassert $(cat ./pkg/testutils/lint/gcassert_paths.txt | sed 's|^|./pkg/|')
See /tmp/gcassert-1563406990.log for full output.

echo $?
0
```

[1] https://github.com/jordanlewis/gcassert/pull/10
